### PR TITLE
Fix playground to display errors

### PR DESCRIPTION
**Pull request by whitphx**
## Description

The current Playground doesn't display any error but it makes debugging harder (internal post: https://huggingface.slack.com/archives/C02QZLG8GMN/p1719463130950669).
This PR fixes it.

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: js/_website/src/lib/components/DemosLite.svelte [hunk 1]
```diff
@@ -82,22 +84,18 @@
...
+	$: requirementsStr = JSON.stringify(requirements); // Use the stringified version to trigger reactivity only when the array values actually change, while the `requirements` object's identity always changes.
 
 	$: if (mounted) {
-		if (timeout) {
-			clearTimeout(timeout);
-		}
-		timeout = setTimeout(() => {
-			update(code, requirements);
-		}, 1000);
+		debounced_run_code && debounced_run_code(code);
```
Comment (whitphx) [inline js/_website/src/lib/components/DemosLite.svelte:90]: Refactoring: when the code is changed, call only `run_code` without `install()`.

File: js/_website/src/lib/components/DemosLite.svelte [hunk 2]
```diff
@@ -52,22 +63,13 @@
...
 			appMode: true
 		});
+		const debounce_timeout = 1000;
+		debounced_run_code = debounce(controller.run_code, debounce_timeout);
+		debounced_install = debounce(controller.install, debounce_timeout);
+
 		mounted = true;
 	});
 
-	function update(code: string, requirements: string[]) {
```
Comment (whitphx) [inline js/_website/src/lib/components/DemosLite.svelte:58]: tbh, I couldn't get why this double function calling could suppress errors, but the problem was fixed as a result of the refactoring.

File: js/_website/src/lib/components/DemosLite.svelte [hunk 3]
```diff
@@ -82,22 +84,18 @@
...
+		debounced_run_code && debounced_run_code(code);
+	}
+	$: if (mounted) {
+		debounced_install && debounced_install(JSON.parse(requirementsStr));
 	}
 
 	let position = 0.5;
 
 	let fullscreen = false;
-	function make_full_screen() {
```
Comment (whitphx) [inline js/_website/src/lib/components/DemosLite.svelte:98]: Refactoring: This method is not used.

File: demo/hello_world/run.py [hunk 4]
```diff
@@ -4,6 +4,6 @@ def greet(name):
     return "Hello " + name + "!"
 
 demo = gr.Interface(fn=greet, inputs="textbox", outputs="textbox")
-    
+
```
Comment (whitphx) [inline demo/hello_world/run.py:7]: nit: This change is not related to the problem, but just formatting the code.

**Comment by whitphx** [comment]
@abidlabs @aliabd There might be a discussion that displaying an error during editing the code is annoying and it should be suppressed somehow, but I couldn't find it. Do you remember something? Perhaps we should add some "smarter" error notification that doesn't annoy users?

**Comment by aliabd** [comment]
@whitphx I also remember some discussion about this. But I think the delay is enough. It used to be that the code would rerun on every change so it was annoying seeing the syntax errors all the time. But with the delay it's really fine. 

I'm open to ideas though

**Review by aliabd — APPROVED** [review]
Changes look great! thanks @whitphx

**Pull request merged**